p2p/src/simulated/network.rs

```diff
@@ -36,43 +43,34 @@ type Task = (
     oneshot::Sender<Vec<PublicKey>>,
 );
 
+/// Configuration for the simulated network.
+pub struct Config {
+    /// Registry for prometheus metrics.
+    pub registry: Arc<Mutex<Registry>>,
+
+    /// Maximum size of a message that can be sent over the network.
+    pub max_size: usize,
+}
```

Comment:
nit: I added `max_size` here, otherwise I just moved this from below so that the `impl` of `Network` could follow directly from the struct definition